frontend/test/src/LiveMessageManager.spec.js

```diff
@@ -4,25 +4,45 @@ import {
   LIVE_MESSAGE_STATUS
 } from '../../src/util/LiveMessageManager.js'
 
+import { enforceOptions } from 'broadcast-channel'
+
+// RJ - 2020-09-13 - NOTE: Makes the tests run faster.
+// See https://github.com/pubkey/broadcast-channel#enforce-a-options-globally
+enforceOptions({ type: 'simulate' })
+
 const apiUrl = 'http://localhost/api'
 const userId = 1
 
+// RJ - 2020-09-13 - NOTE: The next two methods are used to track status changes
+// in this test suit.
+
+const setStatusOrig = LiveMessageManager.prototype.setStatus
+LiveMessageManager.prototype.setStatus = function (status) {
+  setStatusOrig.call(this, status)
+
+  const resolve = (this.statuscallbacks && this.statuscallbacks.shift())
+  resolve && resolve()
+}
+
+LiveMessageManager.prototype.onstatuschange = function (callback) {
+  if (!this.statuscallbacks) this.statuscallbacks = []
+  this.statuscallbacks.push(callback)
+}
+
+// RJ - 2020-09-13 - NOTE: We don't want the election of a leader to take place here
+// We directly open an EventSource connection. We are alone.
+LiveMessageManager.prototype.electLeader = LiveMessageManager.prototype.openEventSourceConnection
+
 const openedManager = (heartbeatTimeOut, reconnectionInterval) => {
   const manager = new LiveMessageManager(heartbeatTimeOut, reconnectionInterval)
-  manager.openLiveMessageConnection(userId, apiUrl)
+  manager.openLiveMessageConnection(userId, apiUrl, true)
   const mockEventSource = manager.eventSource
   mockEventSource.emitOpen()
   return { manager, mockEventSource }
 }
 
-const waitForStatus = async (manager, status, timeout = 25) => {
-  // NOTE SG 2020-07-03 - have to wait a small amount as manager status changes
-  // can be done via a setTimeout
-  var retries = 10
-  while (manager.status !== status && retries > 0) {
-    await new Promise(resolve => setTimeout(resolve, timeout / 10))
-    retries--
-  }
+const waitForStatus = async (manager) => {
+  await new Promise(resolve => { manager.onstatuschange(resolve) })
```

Comment:
I'm not sure I understand the purpose awaiting a new Promise just to give the callback to onstatuschange.
Can you explain the behavior of this please ?

Reply:
I am waiting for the Promise to resolve. The code continues executing as soon as the promise resolves.

Reply:
shouldn't .onstatuschange() be asynchronous then ? That way you wouldn't have to set a callback mechanism

Reply:
Mhm, yes it is, and I did. For `onstatuschange`, resolve is just a regular callback here. I'm using it in a Promise, but it does not know about this.

Reply:
I turned this function into an asynchronous function and used camel case. Does it looks good to you @Skylsmoi?